apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-previewing-theme.ts

```diff
@@ -22,38 +23,37 @@ const getThemeType = ( theme?: Theme ) => {
 };
 
 export const usePreviewingTheme = () => {
-	const previewingThemeSlug = currentlyPreviewingTheme();
-	const previewingThemeId =
-		( previewingThemeSlug as string )?.split( '/' )?.[ 1 ] || previewingThemeSlug;
-	const [ previewingThemeName, setPreviewingThemeName ] = useState< string >(
-		previewingThemeSlug as string
-	);
-	const [ previewingThemeType, setPreviewingThemeType ] = useState< string >();
+	const previewingThemeSlug = useSelect( ( select ) => {
+		// Subscribe to the core store, so that previewingThemeSlug can be updated when the theme is activated.
+		// We need this hack because the currentlyPreviewingTheme() function is not a hook.
+		select( 'core' );
+		return currentlyPreviewingTheme();
```

Comment:
We use `isResolving( 'activateTheme' )` to check whether the previewing theme is activating. Maybe we can use the same trick?

In addition, Gutenberg uses [history.replace](https://github.com/WordPress/gutenberg/blob/f14c509b1d3ea0f8fa4a79202d62c2c1a41550e1/packages/edit-site/src/utils/use-activate-theme.js#L41) to update the query striing. Therefore, I guess we can use the location returned by the [useLocation](https://github.com/WordPress/gutenberg/blob/f14c509b1d3ea0f8fa4a79202d62c2c1a41550e1/packages/router/src/router.js#L19) hook to detect the changes.

WDYT?

Reply:
> We use isResolving( 'activateTheme' ) to check whether the previewing theme is activating. Maybe we can use the same trick?

Could you elaborate the trick? `isResolving` is a selector from the `core` store, which we already subscribe here 😄 

> In addition, Gutenberg uses [history.replace](https://github.com/WordPress/gutenberg/blob/f14c509b1d3ea0f8fa4a79202d62c2c1a41550e1/packages/edit-site/src/utils/use-activate-theme.js#L41) to update the query striing. Therefore, I guess we can use the location returned by the [useLocation](https://github.com/WordPress/gutenberg/blob/f14c509b1d3ea0f8fa4a79202d62c2c1a41550e1/packages/router/src/router.js#L19) hook to detect the changes.

I also considered this but it requires `@wordpress/router`, which is not installed by the `wpcom-block-editor` currently. Does it make sense to add this package for this purpose?

Reply:
I do not fully understand Arthur's intended meaning, but, for reference, I'm sharing the PR that started using `isResolving( 'activateTheme' )`. 🙂  https://github.com/WordPress/gutenberg/pull/55658

Reply:
> Could you elaborate the trick? isResolving is a selector from the core store, which we already subscribe here 😄

Instead of subscribing the core store, we can subscribe that value to trigger the update. Subscribing the core store may trigger the update all the time 🤔

> Does it make sense to add this package for this purpose?

I think it's fine 🙂